### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by IoIxD**
**Clarify to the user an additional step they have to do to compile for WASM.**
We cannot set environment flags to influence the compile process, for some reason. `env::set_var` only sets them in regards to the linker process, and the `println!` directive doesn't seem to work at all on my machine (although, this PR does use it in case it works for somebody).

So the build script now gives the user the `EMCC_CFLAGS` variable that they have to set.

 
>  You must set the following environment variables yourself to compile for WASM. We are sorry for the inconvienence; this will be fixed in 5.1.0.
>  Prefix your command with this (you may want to make a build script for obvious reasons...): EMCC_CFLAGS="-sUSE_GLFW=3 -sGL_ENABLE_GET_PROC_ADDRESS -sWASM=1 -sALLOW_MEMORY_GROWTH=1 -sWASM_MEM_MAX=512MB -sTOTAL_MEMORY=512MB -sABORTING_MALLOC=0 -sASYNCIFY -sFORCE_FILESYSTEM=1 -sASSERTIONS=1 -sERROR_ON_UNDEFINED_SYMBOLS=0 -sEXPORTED_RUNTIME_METHODS=ccallcwrap""

As the error message says, this would only be for 5.0. 5.1 drops emscripten entirely, which would be a bad thing but in this case it will become a blessing as we no longer have to put up with emscripten.

Also:
* Move the log callback code over to C. 
* Renamed the `.ready` functions in the new audio structs to match their raylib counterparts
* Brought back PhantomData for RaylibAudio so that the struct is and !Send and !Sync

### CODE REVIEW

**Comment by IoIxD** [comment]
Accidentally deleted the branch while trying to delete some other ones, oops

File: raylib-sys/binding/utils_log.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#define MAX_TRACELOG_BUFFER_SIZE 128 // As defined in utils.c from raylib
+
+void rayLogWrapperCallback(char *logType, const char *text, va_list args)
+{
+	char buffer[MAX_TRACELOG_BUFFER_SIZE] = {0};
+
+	vsprintf(buffer, text, args);
+
+	custom_trace_log_callback(logType, buffer, strlen(buffer));
```
[the remark this answers is not in the document]
Comment (TSnake41) [reply raylib-sys/binding/utils_log.c:22]: just be careful if the number is larger than buffer size

File: raylib/src/core/callbacks.rs [hunk 2]
```diff
@@ -179,14 +171,13 @@ impl RaylibHandle {
...
     ) -> Result<(), SetLogError> {
-        safe_callback_set_func!(
-            cb,
-            trace_log_callback,
-            set_trace_log_callback,
-            ffi::SetTraceLogCallback,
-            custom_trace_log_callback,
-            "trace log"
-        );
+        if let None = trace_log_callback() {
```
Comment (TSnake41) [inline raylib/src/core/callbacks.rs:174]: if trace_log_callback().is_none() ?

Comment (TSnake41) [reply raylib/src/core/callbacks.rs:174]: and overriding the tracelogcallback may be acceptable, not sure we want to prevent it

**Pull request merged**